src/bot/update_handler.rs

```diff
@@ -0,0 +1,97 @@
+use super::commands::BotCommand;
+use super::commands::CommandProcessor;
+use crate::bot::telegram_client;
+use crate::config::Config;
+use frankenstein::Update;
+use frankenstein::UpdateContent;
+use std::str::FromStr;
+use std::thread;
+
+pub struct UpdateHandler {}
+
+impl UpdateHandler {
+    pub fn start() {
+        let mut api = telegram_client::api().clone();
+        let thread_pool = rayon::ThreadPoolBuilder::new()
+            .num_threads(Config::commands_db_pool_number() as usize)
+            .build()
+            .unwrap();
+
+        log::info!("Starting the El Monitorro bot");
+
+        let interval = std::time::Duration::from_secs(1);
+
+        loop {
+            while let Some(update) = api.next_update() {
+                thread_pool.spawn(move || match update.content {
+                    UpdateContent::CallbackQuery(_) => {
+                        Self::process_callback_query(update);
+                    }
+                    UpdateContent::Message(_) | UpdateContent::ChannelPost(_) => {
+                        Self::process_message_or_channel_post(update);
+                    }
+                    _ => (),
+                });
+            }
+
+            thread::sleep(interval);
+        }
+    }
+
+    fn process_message_or_channel_post(update: Update) {
+        let message = match update.content {
+            UpdateContent::Message(message) => message,
+            UpdateContent::ChannelPost(channel_post) => channel_post,
+            _ => return,
+        };
+
+        if let Some(owner_id) = Self::owner_telegram_id() {
+            if message.from.is_none() {
+                return;
+            }
+
+            if message.from.as_ref().unwrap().id as i64 != owner_id {
+                return;
+            }
+        }
+
+        let text = message.text.clone();
+
+        if text.is_none() {
+            return;
+        }
+
+        let command = BotCommand::from_str(&text.unwrap()).unwrap();
+
+        CommandProcessor::builder()
+            .message(message)
+            .command(command)
+            .build()
+            .process()
+    }
+
+    fn process_callback_query(update: Update) {
+        let query = match update.content {
+            UpdateContent::CallbackQuery(callback_query) => callback_query,
+            _ => return,
+        };
+
+        let text = query.data.clone();
+
+        if text.is_none() || query.message.is_none() {
+            return;
+        }
+
+        let command = BotCommand::from_str(&text.unwrap()).unwrap();
```

Comment:
`?` operator ?

Reply:
any string can be converted to BotCommand without errors